pkgs/test_core/lib/src/runner.dart

```diff
@@ -73,8 +73,14 @@ class Runner {
   /// Sinks created for each file reporter (if there are any).
   final List<IOSink> _sinks;
 
-  /// Creates a new runner based on [configuration].
-  factory Runner(Configuration config) => config.asCurrent(() {
+  /// Creates a new runner based on [configuration] with stdout as the standard reporter.
+  factory Runner(Configuration config) =>
+      Runner.withCustomOutputStream(config, stdout);
+
+  /// Creates a new runner based on [configuration] with a custom standard reporter.
+  factory Runner.withCustomOutputStream(
```

Comment:
How would you use this as a consumer of the test package?

Reply:
It is used like this:
```dart
  var inn = File('exec.s').openRead();
  var out = File('exec.s').openWrite();
  var process = exec.Runner.withCustomOutputStream(c, out);
  await process.run();
  await out.close();
  inn.listen((event) {
    // processing events
  });
```

Reply:
The logic contained under `lib/src` is intended to be private to the package. Consumers of this package should not have access to these APIs.

Reply:
Thanks for your input ! 
Is it fine if I expose class that controls Runner ?

Reply:
The supported way to write a reporter output is with `--file-reporter`. For instance to write the json reporter to a file called `report` `pub run test --file-reporter json:report`.

We don't want to expose this class.

Reply:
What I want to achieve is communicate with the tests, not parsing a file.
Doing so is quite easy with the Runner.
I can understand why you don't want to expose the Runner, but using `pub run test --file-reporter` is not solving my problem.
Solving this problem would enable me to push the tool I coded, which would be a good contribution to the dart community.

Reply:
An API to run tests from Dart code isn't something we currently support or plan to support in the short term. Unfortunately there are too many risks with exposing an API surface area that we may be forced to change in the future. In the very long term we may be able to evolve `test_core` in that direction, but it's not on our roadmap.

Reply:
First, I want to thank you for dedicating time explaining the reasons and doing this review.
I thought the same, exposing it IS complicated compared to the gains, so I totally understand you.
Feel free to close the MR, I don't think this will go your way, at least for now.
Since I really want to release my tool, I'll just fork the test repo and use it for my project, assuming this risk for as long as necessary, and will merge back to this repository the day it decides to expose an API.